packages/cdktf/lib/terraform-stack.ts

```diff
@@ -85,12 +85,13 @@ export class TerraformStack extends Construct {
    *
    * @param tfElement The element for which the logical ID is allocated.
    */
-  protected allocateLogicalId(tfElement: TerraformElement): string {
-    const node = tfElement.constructNode
+  protected allocateLogicalId(tfElement: TerraformElement | Node): string {
+    const node =  "constructNode" in tfElement ? tfElement.constructNode : tfElement;
+    const stack = "stack" in tfElement ? tfElement.stack : this;
```

Comment:
Probably more standard Typescript to have something along the lines of:
```typescript
if (tfElement instanceof TerraformElement) {
...
}
else {
...
}
```

Reply:
Nice, I thought that would be too simple :D